system/include/emscripten/em_macros.h

```diff
@@ -14,3 +14,25 @@
 #else
 #define EM_IMPORT(NAME)
 #endif
+
+/*
+ * EM_JS_DEPS: Use this macro to declare indirect dependencies on JS symbols.
+ * The first argument is just unique name for the set of dependencies, that
+ * second argument is a C string that lists JS library symbols in the same way
+ * they would be specified in the DEFAULT_LIBRARY_FUNCS_TO_INCLUDE command line
+ * setting.
+ *
+ * For example if your code contains EM_ASM or EM_JS block that make use of the
+ * allocate, and stackSave JS library functions then you might write this in
+ * your library source code:
+ *
+ *   EM_JS_DEPS(mylib_dep, "$allocate,$stackSave");
+ *
+ * The emscripten linker will then pick this up and make sure those symbols get
+ * included in the JS support library.
+ */
```

Comment:
Maybe worth clarifying this is tied to the file/compilation unit?

Reply:
Done